components/popover/stories/template.js

```diff
@@ -16,146 +20,284 @@ export const Template = ({
-					const popover = document.querySelector(`#${id}`);
-					if (!popover) return [];
+			// Do nothing if the height and width are the same; prevent loops
+			if (!shouldChange) return;
 
-					const transforms = [];
-					const additionalStyles = {};
-					const triggerXCenter = (rect.left + rect.right) / 2;
-					const triggerYCenter = (rect.top + rect.bottom) / 2;
-					const popWidth = popover.offsetWidth ?? 0;
-					const popHeight = popover.offsetHeight ?? 0;
-					const textDir = getComputedStyle(document.querySelector("html")).direction;
-					let x, y;
-					let xOffset = "+ 0px";
-					let yOffset = "+ 0px";
-					if (position.includes("top") || position.includes("bottom") && !(position.includes("-top") || position.includes("-bottom"))) {
-						x = triggerXCenter - (popWidth > 0 ? popWidth / 2 : popWidth);
-					}
-					if (position.includes("left") || position.includes("right")) {
-						y = triggerYCenter - (popHeight > 0 ? popHeight / 2 : popHeight);
-					}
-					if (position.includes("top") && !position.includes("-top")) {
-						y = rect.top - popHeight;
-						yOffset = withTip
-							? "- (var(--spectrum-popover-pointer-height) + var(--spectrum-popover-animation-distance) - 1px)"
-							: "- var(--spectrum-popover-animation-distance)";
-					}
- else if (position.includes("bottom") && !position.includes("-bottom")) {
-						y = rect.bottom;
-						yOffset = "+ (var(--spectrum-popover-animation-distance))";
-					}
- else if (position.includes("left")) {
-						if (textDir == "rtl") {
-							x = rect.right;
-							xOffset = withTip ? "+ 0px" : "+ var(--spectrum-popover-animation-distance)";
-						}
- else {
-							x = rect.left - popWidth;
-							xOffset = withTip
-								? "- ((var(--spectrum-popover-pointer-width) / 2) + var(--spectrum-popover-animation-distance) - 2px)"
-								: "- var(--spectrum-popover-animation-distance)";
-						}
-					}
- else if (position.includes("right")) {
-						if (textDir == "rtl") {
-							x = rect.left - popWidth;
-							xOffset = withTip
-								? "- ((var(--spectrum-popover-pointer-width) / 2) + var(--spectrum-popover-animation-distance) - 2px)"
-								: "- var(--spectrum-popover-animation-distance)";
-						}
- else {
-							x = rect.right;
-							xOffset = withTip ? "+ 0px" : "+ var(--spectrum-popover-animation-distance)";
-						}
-					}
+			// Write the actual height and width of the popover to the CSS custom properties
+			updateArgs({
+				popoverWidth: parseInt(rect.width, 10),
+				popoverHeight: parseInt(rect.height, 10),
+			});
+		}, 500);
+	});
 
-					if (x) transforms.push(`translateX(calc(var(--flow-direction) * calc(${parseInt(x, 10)}px ${xOffset})))`);
-					if (y) transforms.push(`translateY(calc(${y}px ${yOffset}))`);
+	if (!skipAlignment) {
+		switch (position) {
+			case "top":
+				popoverWrapperStyles["inline-size"] = "var(--spectrum-popover-width)";
+				popoverAlignment["inset-block-end"] = "100%";
```

Comment:
I removed 2 lines from these test cases about "inset-inline-start" being set to "0". Something about those 2 lines was uncentering the top and bottom positioning of the popover. I was wondering if maybe it had to do with the additional horizontal padding popover has now? 


https://github.com/adobe/spectrum-css/blob/196ef4c0dd64f9c6c3803197453cd8dc28cc4ff4/components/popover/stories/template.js#L70
https://github.com/adobe/spectrum-css/blob/196ef4c0dd64f9c6c3803197453cd8dc28cc4ff4/components/popover/stories/template.js#L95

Reply:
This is a good callout/observation! After looking into it, I concur that it looks like it's because the horizontal padding is added, but I think we might be able to keep these two lines as they were if we make an adjustment to the `popoverWidth` and `popoverHeight` that are being set in the template args and the stories.js. Those dimensions were being used to match the calculated popover size but don't look like they are anymore, so either way we probably want to update them to account for that padding.

(Adding some screenshots, in case it makes more sense with illustrations:)

Here's the old popover (on main) with 89x142 dimensions seen in the custom props in the wrapper and also for .spectrum-Popover:
![image](https://github.com/user-attachments/assets/46d3787e-6e64-4e8f-be7b-8ff974d79b0b)

And here's the branch, but the dimensions don't match the custom props:
![image](https://github.com/user-attachments/assets/7841c5f2-1f2b-4be8-a1da-78c13e434631)

Reply:
Ok. I added the additional padding (16px I guess, on both sides) to the args, and like you said, that fixes the off-centering. 🥳 The `inset-inline-start` lines were added back in! Those args are now set to 105px and 158px.

I'll try to get with @jawinn so I can understand more about the `popoverWidth` and `popoverHeight`. Are they supposed to be like min-width & height values? I'm definitely seeing smaller popovers, so those must be coming from the event listener. I'm also curious if i need to adjust the new combobox [popoverHeight](https://github.com/adobe/spectrum-css/blob/8382527c9d1aa0889829ef9868b609f3b018b9b3/components/combobox/stories/template.js#L110)/[popoverWidth](https://github.com/adobe/spectrum-css/blob/8382527c9d1aa0889829ef9868b609f3b018b9b3/components/combobox/stories/template.js#L143) I introduced (which were on main). Do each of those sizes need to be adjusted for the extra padding?

<img width="336" alt="Screenshot 2024-11-21 at 3 59 35 PM" src="https://github.com/user-attachments/assets/2872b6e9-d499-473a-a6de-903ee79cf536">
<img width="340" alt="Screenshot 2024-11-21 at 3 40 37 PM" src="https://github.com/user-attachments/assets/30d7ea6c-0fb0-42c2-ac19-7b12754e1c21">

Reply:
From what I can tell about `popoverWidth` and `popoverHeight`, they are to define the actual fixed dimensions of the popover wrapper, I think to help with our example positioning. Either passed as args or the event listener tries to determine it dynamically. It appears that `popoverWrapperStyles["inline-size"] = "var(--spectrum-popover-width)";` is used in the alignment code of the template, and perhaps the height custom property is not being used. Perhaps we can look at this if there are still questions here.